src/ff1.rs

```diff
@@ -68,8 +65,12 @@ impl Radix {
                 log_radix: u8::try_from(log_radix).unwrap(),
             }
         } else {
-            use libm::{ceil, log10};
-            let min_len = ceil(LOG_10_MIN_NS_DOMAIN_SIZE / log10(f64::from(radix))) as u32;
+            let mut min_len = 1u32;
+            let mut domain = radix;
+            while domain < MIN_NS_DOMAIN_SIZE {
+                domain *= radix;
```

Comment:
```suggestion
                // This multiplication cannot overflow because (MIN_NS_DOMAIN_SIZE-1) * (1 << 16) fits in u64.
                domain *= radix; 
```